### PR TITLE
Login

### PR DESCRIPTION
**Pull request by darmhoo**
This is login

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_API_URL="http://localhost:8000/api"
```
Comment (teebarg) [inline .env:1]: Cant we just have API_URL?

File: .gitignore [hunk 2]
```diff
@@ -1,12 +1,3 @@
-# Logs
-logs
```
Comment (teebarg) [inline .gitignore:2]: Why did you delete line `1 `to `8`?

File: package.json [hunk 3]
```diff
@@ -11,9 +11,18 @@
     "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
```
Comment (teebarg) [inline package.json]: why are we adding `@fortawesome` dependency?

File: package.json [hunk 4]
```diff
@@ -11,9 +11,18 @@
...
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/vue-fontawesome": "^3.0.2",
+    "axios": "^1.1.3",
+    "mosha-vue-toastify": "^1.0.23",
     "pinia": "^2.0.23",
+    "vee-validate": "^4.7.2",
     "vue": "^3.2.41",
-    "vue-router": "^4.1.5"
+    "vue-router": "^4.1.5",
+    "yup": "^0.32.11"
```
Comment (teebarg) [inline package.json]: what is `yup` for?

File: package.json [hunk 5]
```diff
@@ -11,9 +11,18 @@
...
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-brands-svg-icons": "^6.2.0",
+    "@fortawesome/free-regular-svg-icons": "^6.2.0",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/vue-fontawesome": "^3.0.2",
+    "axios": "^1.1.3",
+    "mosha-vue-toastify": "^1.0.23",
     "pinia": "^2.0.23",
+    "vee-validate": "^4.7.2",
```
Comment (teebarg) [inline package.json]: let's use `https://formkit.com/` instead

File: package.json [hunk 6]
```diff
@@ -11,9 +11,18 @@
     "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.2.0",
+    "@fortawesome/free-brands-svg-icons": "^6.2.0",
+    "@fortawesome/free-regular-svg-icons": "^6.2.0",
+    "@fortawesome/free-solid-svg-icons": "^6.2.0",
+    "@fortawesome/vue-fontawesome": "^3.0.2",
+    "axios": "^1.1.3",
+    "mosha-vue-toastify": "^1.0.23",
```
Comment (teebarg) [inline package.json:17]: remove this

File: src/api/client.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+const auth = useAuthStore();
+export const get = (url: string) =>
+  instance({
+    method: "GET",
+    url,
+    headers: { Authorization: `Bearer ${auth.user?.api_token}` },
+  });
+export const post = (url: string, data: object) => {
+  console.log(auth);
```
Comment (teebarg) [inline src/api/client.ts]: remove console logs

File: .env (hunk 1 above)
Comment (darmhoo) [reply .env:1]: I dont think it will work according to the documentation for dev. I sholud have placed this in gitignore

File: package.json (hunk 4 above)
Comment (darmhoo) [reply package.json]: Yup works with vee-validate to give chainable validation with the composition api

File: package.json (hunk 3 above)
Comment (darmhoo) [reply package.json]: for easy icons

Comment (teebarg) [reply package.json]: i don't think we need a package to manage icons

File: package.json (hunk 4 above)
Comment (teebarg) [reply package.json]: Let's us use Formkit for ours forms, so we don't need this.

File: .env (hunk 1 above)
Comment (teebarg) [reply .env:1]: Can you check? it should

Comment (darmhoo) [reply .env:1]: We ave the option of customizing it. But we hhave to use a pefix

Comment (darmhoo) [reply .env:1]: Instead of VITE_ We ccan use VUE_APP

Comment (darmhoo) [reply .env:1]: https://vitejs.dev/config/shared-options.html#envprefix

File: package.json (hunk 6 above)
Comment (darmhoo) [reply package.json:17]: it makes notification easy
